core/src/test/java/org/elasticsearch/common/settings/SettingsTests.java

```diff
@@ -68,6 +68,17 @@ public void testReplacePropertiesPlaceholderSystemProperty() {
         assertThat(settings.get("setting1"), equalTo(value));
     }
 
+    public void testReplacePropertiesPlaceholderSystemPropertyList() {
+        String value = System.getProperty("java.home");
+        assertFalse(value.isEmpty());
```

Comment:
I don't think this will be needed if you take the suggestion below, but in general `assertFalse` and `assertTrue` should be avoided because they do not give good error messages (a failing expectation would only say `AssertionError`). Here, we could use `assertThat(value, not(isEmptyOrNullString())` as then the expectation would say

```
Expected: not (null or an empty string)
     but: was ""
```

which is a lot more helpful!

Reply:
thank you, this is very helpful to know